### ISSUE TITLE
Question around Zsh plugins management

### ISSUE THREAD

**Issue by crazy-matt** [kclejeune/system#217]
Hi Kennan,

I've seen your recent improvements, nice nice! Especially the move to Cirrus for aarch64 bulid support.
I had a question for you about these lines: https://github.com/kclejeune/system/blob/85054d79344284ec5d4c68a6bb0cad445963730c/modules/home-manager/shell.nix#L53

Do you see any downside in using a fast plugin manager like [sheldon](https://search.nixos.org/packages?channel=unstable&show=sheldon&from=0&size=50&sort=relevance&type=packages&query=sheldon) compared to wrapping the deployment of plugins in `shell.nix` like you did?

**Comment by crazy-matt** [comment]
Rephrasing the question, do you see a benefit in using your wrapper instead of an external plugin manager and passing `~/.config/sheldon/plugins.toml` as a dotfile?

**Comment by kclejeune** [comment]
> Rephrasing the question, do you see a benefit in using your wrapper instead of an external plugin manager and passing `~/.config/sheldon/plugins.toml` as a dotfile?

Good question! I'm not familiar with Sheldon specifically, but I previously used zplug to fill a similar role. I ended up dropping it mainly because I didn't really feel the need for plug-in manager since all my plugins are in nixpkgs, and at that point it's easy enough to source them directly and version them from nixpkgs with the rest of my configuration. 

Sheldon seems like a cool tool, but I don't know how well it fits the "nix philosophy" to pull plugins from somewhere else.

**Comment by crazy-matt** [comment]
Yeah it doesn't seem to be nix philosophy proof. But I'll give it a try just for the sake of training myself. And also for the infinite capabilities of sheldon. As soon as plugin is released, you can test it with sheldon without waiting for a nix implementation.

**Comment by kclejeune** [comment]
> Yeah it doesn't seem to be nix philosophy proof. But I'll give it a try just for the sake of training myself. And also for the infinite capabilities of sheldon. As soon as plugin is released, you can test it with sheldon without waiting for a nix implementation.
> 

Sure - again, there's nothing inherently wrong with it, my zsh config just isn't really complicated enough to warrant it. I think this is similar to using a regular vim config and a plugin manager within a nix config, which is certainly still a popular thing to do.

**Comment by crazy-matt** [comment]
I'm stuck on something at the moment. Sheldon is packaged only for linux platforms. So I started deploying it via brew with nix-darwin. And I realise that your cool wrapper is expecting the package for both platforms, as we're under home-manager.

**Comment by crazy-matt** [comment]
It might sounds a noob question (I'm a complete noob on Nix), how would you manage this case?

**Comment by crazy-matt** [comment]
Should I modify the mkZshPlugin function to be able to pass a specific source for darwin platforms with a ternary operator and isDarwin?

**Comment by kclejeune** [comment]
> Should I modify the mkZshPlugin function to be able to pass a specific source for darwin platforms with a ternary operator and isDarwin? 

If you're using a separate plugin manager then ideally you wouldn't be configuring your shell with home manager at all. You either grab plugins using a plugin manager or using nixpkgs and the mkZshPlugin wrapper, but not both.
